Sources/OpenTelemetryXRay/IDGenerator.swift

```diff
@@ -12,19 +12,20 @@
 //===----------------------------------------------------------------------===//
 
 import struct Dispatch.DispatchWallTime
-@_exported import OpenTelemetry
+import NIOConcurrencyHelpers
+@_exported import OTel
 
 /// Generates trace and span ids using a `RandomNumberGenerator` in an X-Ray compatible format.
 ///
 /// - SeeAlso: [AWS X-Ray: Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader)
-public struct XRayIDGenerator: OTelIDGenerator {
-    private let getCurrentSecondsSinceEpoch: () -> UInt32
-    private var randomNumberGenerator: RandomNumberGenerator
+public struct XRayIDGenerator<NumberGenerator: RandomNumberGenerator & Sendable>: OTelIDGenerator {
+    private let getCurrentSecondsSinceEpoch: @Sendable () -> UInt32
+    private let randomNumberGenerator: NIOLockedValueBox<NumberGenerator>
 
     /// Initialize an X-Ray compatible `OTelIDGenerator` backed by the given `RandomNumberGenerator`.
     ///
     /// - Parameter randomNumberGenerator: The `RandomNumberGenerator` to use, defaults to a `SystemRandomNumberGenerator`.
```

Comment:
Given that this initializer doesn't have a default `RandomNumberGenerator` anymore, could we remove this bit of documentation, and instead document the public init you added in the extension below too?